Pod/Illuminator/IlluminatorTestProgress.swift

```diff
@@ -9,6 +9,42 @@
 import Foundation
 import XCTest
 
+/*
+ * This protocol allows custom handling of results.  For example, screenshots of failure states or desktop notifications
+ * isPass and isFail can both be false -- it indicates a "Flagging" state.  They are guaranteed to not both be true
+ */
+public protocol IlluminatorTestResultHandler {
+    associatedtype AbstractStateType
```

Comment:
I understand that, but two things:
- you  are making all tests in a test case be single state type 
- state type doesnt have any requirements as a generic constraint on caller side so that side doesnt care

so if you make it a generic callback on IlluminatorTestProgress you will get your type info passed back but without coupling it to test case